sql/src/main/java/io/crate/metadata/Schemas.java

```diff
@@ -122,29 +122,49 @@ public TableInfo resolveTableInfo(QualifiedName ident, Operation operation, User
                     }
                 }
             }
+            if (tableInfo == null) {
+                SchemaInfo currentSchema = schemas.get(searchPath.currentSchema());
+                if (currentSchema == null) {
+                    throw new RelationUnknown(tableName);
+                } else {
+                    throw RelationUnknown.of(tableName, getSimilarTables(user, tableName, currentSchema.getTables()));
+                }
+            }
         } else {
             schemaInfo = schemas.get(identSchema);
             if (schemaInfo == null) {
                 throw SchemaUnknownException.of(identSchema, getSimilarSchemas(user, identSchema));
             } else {
                 tableInfo = schemaInfo.getTableInfo(tableName);
+                if (tableInfo == null) {
+                    throw RelationUnknown.of(ident.toString(), getSimilarTables(user, tableName, schemaInfo.getTables()));
+                }
             }
         }
-        if (tableInfo == null) {
-            throw new RelationUnknown(ident.toString());
-        }
         Operation.blockedRaiseException(tableInfo, operation);
         return tableInfo;
     }
 
+    private static List<String> getSimilarTables(User user, String tableName, Iterable<TableInfo> tables) {
+        LevenshteinDistance levenshteinDistance = new LevenshteinDistance();
+        ArrayList<String> candidates = new ArrayList<>();
+        for (TableInfo table : tables) {
+            if (user.hasAnyPrivilege(Privilege.Clazz.TABLE, table.ident().fqn())) {
+                String candidate = table.ident().name();
+                if (levenshteinDistance.getDistance(tableName, candidate) > 0.7f || tableName.equalsIgnoreCase(candidate)) {
+                    candidates.add(candidate);
+                }
+            }
+        }
+        return candidates;
```

Comment:
Do you think it is worth to include some scoring for ordering and limit this to maybe max 5 candidates?

Reply:
sounds nice! so users are not overwhelmed with hints.

Shall we also add a change log entry for both this and schema hints?

Reply:
@kovrus pushed a fixup